src/optparse.coffee

```diff
@@ -99,14 +88,22 @@ buildRule = (shortFlag, longFlag, description, options = {}) ->
     isList:       !!(match and match[2])
   }
 
-# Normalize arguments by expanding merged flags into multiple flags. This allows
-# you to have `-wl` be the same as `--watch --lint`.
-normalizeArguments = (args) ->
-  args = args[..]
-  result = []
-  for arg in args
-    if match = arg.match MULTI_FLAG
-      result.push '-' + l for l in match[1].split ''
-    else
-      result.push arg
-  result
+addArgument = (rule, options, value) ->
+  options[rule.name] = if rule.isList
+    (options[rule.name] ? []).concat value
+  else value
+
+tryMatchOptionalArgument = (arg, state, rules) ->
+  for rule in rules
+    if arg in [rule.shortFlag, rule.longFlag]
+      if rule.hasArgument
+        value = state.argsLeft.shift()
+        if not value?
+          throw new Error "#{arg} requires a value, but was the last argument"
+      else
+        value = true
+
+      addArgument(rule, state.options, value)
+      return
```

Comment:
If not, it would eventually leave the for loop and hit the throw statement. This was dealt with previously by keeping a boolean testing whether or not the rule was matched; I thought this was cleaner here since we don't then loop over the rest of the rules if we successfully match one.